Source/HarmonyPatches/Patches_Room/Patch_Room.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection.Emit;
+using HarmonyLib;
+using PrisonLabor.Core.Trackers;
+using Verse;
+
+namespace PrisonLabor.HarmonyPatches.Patches_Rooms
+{
+    [StaticConstructorOnStartup]
+    public static class Patch_Room
+    {
+        static Patch_Room()
+        {
+            var harmony = new Harmony("Harmony_PrisonLabor");
+
+            var mBase = AccessTools.Method(typeof(RoomGroup), nameof(RoomGroup.AddRoom));
+            var mPost = AccessTools.Method(typeof(Patch_Room), nameof(Patch_Room.Postfix));
+
+            harmony.Patch(mBase, new HarmonyMethod(mPost));
+        }
+
+        public static void Postfix(Room room)
+        {
+            lock (Tracked.LOCK_WARDEN)
+            {
+                if (!Tracked.Wardens.Keys.Contains(room.ID))
+                    Tracked.Wardens[room.ID] = new List<int>();
+
+                if (!Tracked.Prisoners.Keys.Contains(room.ID))
+                    Tracked.Prisoners[room.ID] = new List<int>();
+#if DEBUG
+                Log.Message("Added Region " + room.ID);
+#endif
+            }
+        }
+    }
+
+    [HarmonyPatch(typeof(Room))]
+    [HarmonyPatch(nameof(Room.RemoveRegion))]
+    public static class Patch_RoomDrefrenced
+    {
+        static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions)
```

Comment:
Postfix? its more update proof

Reply:
it's an if block.
<img width="757" alt="Screen Shot 2020-07-07 at 1 40 51 PM" src="https://user-images.githubusercontent.com/6241020/86840377-7a9cb700-c057-11ea-9738-0a42521f868c.png">
I'm patching the ```if(region.count == 0)``` block